### PR TITLE
Improvement of the matrix sparsity pattern of DG on single core

### PR DESCRIPTION
**Pull request by yinghe616**
This pull request contains two changes:
1) for single core, call the make_flux_sparsity_pattern for DG with face coupling, which has less nonzero numbers; for multiple cores, call the make_flux_sparsity_pattern for DG with face coupling  at every where, but it allows to use multiple processor.

2) move the DG BP limiter to the solver.cc inside the function call of solve_advection.

@egpuckett

### CODE REVIEW

File: source/simulator/core.cc [hunk 1]
```diff
@@ -1781,20 +1837,13 @@ namespace aspect
           assemble_advection_system (AdvectionField::temperature());
           solve_advection(AdvectionField::temperature());
 
-          if (parameters.use_discontinuous_temperature_discretization
-              && parameters.use_limiter_for_discontinuous_temperature_solution)
-            apply_limiter_to_dg_solutions(AdvectionField::temperature());
-
```
Comment (bangerth) [inline source/simulator/core.cc]: This change is independent of the other one. Can you separate this into a different pull request?

Comment (yinghe616) [inline source/simulator/core.cc]: OK. I will do.

File: source/simulator/core.cc [hunk 2]
```diff
@@ -1043,17 +1043,45 @@ namespace aspect
...
+              face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::always;
+          }
+        if (Utilities::MPI::n_mpi_processes(mpi_communicator) == 1)
+          {
+            DoFTools::make_sparsity_pattern (dof_handler,
+                                             coupling, sp,
+                                             constraints, false,
+                                             Utilities::MPI::
+                                             this_mpi_process(mpi_communicator));
+            DoFTools::make_flux_sparsity_pattern (dof_handler,
```
Comment (tjhei) [inline source/simulator/core.cc]: you don't need to run both functions, just make_flux_sparsity_pattern should do both.

Comment (tjhei) [inline source/simulator/core.cc]: oh, or do you need this because of the constraints?

Comment (yinghe616) [inline source/simulator/core.cc]: Yes, for the constraints. Otherwise, it gives run-error messages.
